wasm2c/wasm-rt-os-unix.c

```diff
@@ -0,0 +1,190 @@
+// Based on
+// https://web.archive.org/web/20191012035921/http://nadeausoftware.com/articles/2012/01/c_c_tip_how_use_compiler_predefined_macros_detect_operating_system#BSD
+// Check for any posix or unix OS
+#if !defined(_WIN32) && (defined(__unix__) || defined(__unix) || \
+                         (defined(__APPLE__) && defined(__MACH__)))
+
+#include "wasm-rt-os.h"
+#include "wasm-rt.h"
+
+#include <errno.h>
+#include <inttypes.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#if defined(__APPLE__) && defined(__MACH__)
+// Macs priors to OSX 10.12 don't have the clock functions. So we will use mac
+// specific options
+#include <mach/mach_time.h>
+#include <sys/time.h>
+#endif
+#include <sys/mman.h>
+#include <unistd.h>
+
+#ifdef VERBOSE_LOGGING
+#define VERBOSE_LOG(...) \
+  { printf(__VA_ARGS__); }
+#else
+#define VERBOSE_LOG(...)
+#endif
+
+size_t os_getpagesize() {
+  return getpagesize();
+}
+
+void* os_mmap(void* hint, size_t size, int prot, int flags) {
+  int map_prot = PROT_NONE;
+  int map_flags = MAP_ANONYMOUS | MAP_PRIVATE;
+  uint64_t request_size, page_size;
+  uint8_t* addr;
+
+  page_size = (uint64_t)os_getpagesize();
+  request_size = (size + page_size - 1) & ~(page_size - 1);
```

Comment:
This is common enough in the code below it probably should be a macro (`ALIGN_UP`?)